index.js

```diff
@@ -44,13 +44,56 @@ function reduce(array) {}
 
 function reduceRight(array) {}
 
-function reverse(array) {}
+function reverse(array) {
+  const newArray = []; //wrote this as array(), getting confused with PHP;
+  for (let i = (array.length-1); i >= 0; i--){ //created infinite loop and issue with i being > than 0 rather than <: for (let i = (array.length-1); i <= array.length; i--){
+    newArray.push(array[i]);
+  }
+  return newArray;
+}
 
-function shift(array) {}
+function shift(array) {
+  if(array.length > 0){
+    array.length = array.length - 1; //didn't know how to reduce the length. I looked a the example in pop
+    return array[0];
+  }else{
+    return undefined;
+  }
+}
 
-function slice(array) {}
+/**
+The slice() method returns a shallow copy of a portion of an array into a new array
+object selected from begin to end (end not included). The original array will not be modified.
+**/
+function slice(array, start, end) {
+  //had to google parseInt to remember how it's called; //also don't know all bases!
+  const newArray = [];
+  if (typeof start === 'undefined' || start === ''){ // could't remember how to check for undefined: if (start == undefined){
+    start = 0; //if here I wrote let start = 0 it then would still be undefined?
+  }
+  start = parseInt(start,10);
```

Comment:
you can check for start index using
```js
if(!start){ // this will be true for 0, undefined, null, ''
start = 0
}

OR this is same as

start = start || 0
```